src/modules/launcher/PowerLauncher/ViewModel/MainViewModel.cs

```diff
@@ -461,16 +462,19 @@ private void QueryResults()
                 _updateSource = currentUpdateSource;
                 var currentCancellationToken = _updateSource.Token;
                 _updateToken = currentCancellationToken;
+                var queryText = QueryText.Trim();
 
-                var query = QueryBuilder.Build(QueryText.Trim(), PluginManager.NonGlobalPlugins);
-                if (query != null)
+                var pluginQueryPair = QueryBuilder.Build(ref queryText, PluginManager.NonGlobalPlugins);
+                if (pluginQueryPair != null)
```

Comment:
Can you please add a check for empty as well ? This would prevent spawning a thread in case pluginQueryPair is empty